packages/devtools_app/test/file_search_test.dart

```diff
@@ -0,0 +1,164 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/debugger/file_search.dart';
+import 'package:devtools_app/src/ui/search.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+
+import 'support/mocks.dart';
+import 'support/wrappers.dart';
+
+void main() {
+  final debuggerController = MockDebuggerController.withDefaults();
+  final autoCompleteController = AutoCompleteController();
+
+  Widget buildFileSearch() {
+    return MaterialApp(
+      home: Scaffold(
+        body: Card(
+          child: FileSearchField(
+              debuggerController: debuggerController,
+              autoCompleteController: autoCompleteController),
+        ),
+      ),
+    );
+  }
+
+  group('File search', () {
+    setUp(() {
+      when(debuggerController.sortedScripts)
+          .thenReturn(ValueNotifier(mockScriptRefs));
+    });
+
+    testWidgetsWithWindowSize(
+        'Search returns expected files', const Size(1000.0, 4000.0),
+        (WidgetTester tester) async {
+      await tester.pumpWidget(buildFileSearch());
+
+      autoCompleteController.search = '';
+      expect(
+          autoCompleteController.searchAutoComplete.value,
+          equals([
+            // Show all results (truncated to 10):
+            'animals/cats/meow.dart',
+            'animals/cats/purr.dart',
+            'animals/dogs/bark.dart',
+            'animals/dogs/growl.dart',
+            'animals/insects/caterpillar.dart',
+            'animals/insects/cicada.dart',
+            'food/catering/party.dart',
+            'food/carton/milk.dart',
+            'food/milk/carton.dart',
+            'travel/adventure/cave_tours_europe.dart',
+          ]),
+          reason: 'Correct search results for empty query.');
+
+      autoCompleteController.search = 'c';
+      expect(
+          autoCompleteController.searchAutoComplete.value,
+          equals([
+            // Exact matches:
+            'animals/cats/meow.dart',
+            'animals/cats/purr.dart',
+            'animals/insects/caterpillar.dart',
+            'animals/insects/cicada.dart',
+            'food/catering/party.dart',
+            'food/carton/milk.dart',
+            'food/milk/carton.dart',
+            'travel/adventure/cave_tours_europe.dart',
+            'travel/canada/banff.dart',
+          ]),
+          reason: 'Correct search results for "c" query.');
```

Comment:
don't worry about supplying a reason unless it is really surprising. For example, here it is clear the query is 'c' so no need to document that.

Reply:
failures will have a line and column clarifying what the failure is.

Reply:
Ah good to know, done!